### DISCUSSION TITLE
Mesh reduction throw access violation exception.

### DISCUSSION THREAD

**Discussion by NielsHarmelink** [bepu/bepuphysics2#229] [Q&A]
When I'm trying to check for a collision, the BepuPhysics.CollisionDetection.MeshReduction.ChildEnumerator throws an access violation exception. I use BepuPhysics for collision checks only, when a collision occurs it returns the collision and will stop the collision checks. Is mesh reduction needed for collision checks only? If Mesh reduction is needed how can I solve this problem? If it's not needed is there a possibility to disable it? I'm currently running on the latest version of 2.3.X.  (I tried to create a simple repo but cannot reproduce it in the simple repo. It only occurs in the program I use for collision checks.

![image](https://user-images.githubusercontent.com/106063532/182152793-80a7b5df-84cd-4738-8bac-f60855cb679f.png)

**Comment by RossNordby** [comment]
You could make this return unconditional: https://github.com/bepu/bepuphysics2/blob/master/BepuPhysics/CollisionDetection/MeshReduction.cs#L321

... but, given that reproducing the issue was difficult, there is probably something else going on that needs to be addressed. If the failure in the original context was also unreliable, my first suspicion would be race conditions (most likely improper asynchronous writes).

**Reply by NielsHarmelink** [reply]
I do not use a thread dispatcher or use the collision check by multiple threads. Maybe it has to do something with a very complex mesh? The mesh has over 1000 indices and around 250 vertexes.

**Reply by RossNordby** [reply]
I don't think mere mesh complexity would explain it- I've tested it on meshes orders of magnitude larger than that. Maybe some weird triangle data could do it.

The best option is probably to try a few more repro attempts. I doubt I can guess the issue.

**Reply by NielsHarmelink** [reply]
You were right the mesh contains weird triangle data, fixed it. but came across another problem. Luckily I could reproduce this problem and created a repro for it.  It throws a null exception when trying to get tree overlaps in mesh reduction because the leaf count is null. When running the repro the error will occur automatically. 

[bepuphysics2-2.3.X.zip](https://github.com/bepu/bepuphysics2/files/9251312/bepuphysics2-2.3.X.zip)
.

**Reply by NielsHarmelink** [reply]
Found the Issue mesh was never set to a value and automatically result in null exception. Strange that i never came across this issue. But I've created a pull request for this fix 😄.

**Reply by RossNordby** [reply]
Good catch, thanks!
